lms/djangoapps/etc_api/views.py

```diff
@@ -116,17 +125,30 @@ def post(self, request):
         or activate enroll with use 'is_active' param.
         """
         data = request.data
+        user_id = data.get('user_id')
+        course_id = data.get('course_id')
+        if not user_id:
+            return Response(data={"user_message": "'user_id' is required parameter."}, status=400)
+
+        if not course_id:
+            return Response(data={"user_message": "'course_id' is required parameter."}, status=400)
+
         try:
-            user = User.objects.get(id=data['user_id'])
+            user = User.objects.get(id=user_id)
         except  User.DoesNotExist:
-            return Response(data={"user_message": "Wrong id User does not exist"}, status=400)
-        course_id = SlashSeparatedCourseKey.from_deprecated_string(data['course_id'])
-        enrollment_obj, __ = CourseEnrollment.objects.update_or_create(
-            user=user,
-            course_id=course_id,
-            defaults={
-                'mode': data['mode'],
-                'is_active': string_to_boolean(data['is_active'])
-            }
-        )
-        return Response(data={'enrollment_id': enrollment_obj.id}, status=status.HTTP_200_OK)
\ No newline at end of file
+            return Response(data={"user_message": "Wrong 'user_id'. User does not exist"}, status=400)
+
+        try:
+            course_id = SlashSeparatedCourseKey.from_deprecated_string(course_id)
+            enrollment_obj, __ = CourseEnrollment.objects.update_or_create(
+                user=user,
+                course_id=course_id,
+                defaults={
+                    'mode': data.get('mode', 'honor'),
+                    'is_active': string_to_boolean(data.get('is_active'))
+                }
+            )
+        except InvalidKeyError:
+            return Response(data={"user_message": "Wrong 'course_id'. Course does not exist"}, status=400)
+
+        return Response(data={'enrollment_id': enrollment_obj.id}, status=status.HTTP_200_OK)
```

Comment:
If you use `HTTP_200_OK` why you don't use `HTTP_400_BAD_REQUEST` in lines L152 and 139

Reply:
I fixed that mistakes.